tests/Agent/UnitTests/Core.UnitTest/DataTransport/HttpCollectorWireWebRequestTests.cs

```diff
@@ -0,0 +1,377 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#if NETFRAMEWORK
+
+using System.Net.Http;
+using System.Net;
+using System;
+using NewRelic.Agent.Configuration;
+using NewRelic.Agent.Core.AgentHealth;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text;
+using NUnit.Framework;
+using System.Threading.Tasks;
+using System.Threading;
+using NewRelic.Agent.Core.DataTransport.Client;
+using NewRelic.Agent.Core.DataTransport.Client.Interfaces;
+using NewRelic.Agent.Core.Exceptions;
+using NewRelic.Agent.Core.Logging;
+using Serilog;
+using Telerik.JustMock;
+
+namespace NewRelic.Agent.Core.DataTransport
+{
+    public class WebRequestTestResponse
+    {
+        public HttpStatusCode StatusCode { get; set; }
+        public string StringContent { get; set; }
+
+        public byte[] ByteArrayContent { get; set; }
+
+        public long ContentLength => ByteArrayContent?.Length ?? StringContent?.Length ?? 0;
+
+        public string ContentEncoding { get; set; }
+    }
+
+    [TestFixture]
+    public class HttpCollectorWireWebRequestTests
```

Comment:
I'm assuming these tests are very similar to the HttpClient tests. Should we try to ensure that we are running the same tests for both? Right now it is easier for the tests to diverge.

Reply:
After looking at the tests, I can do a significant refactor that will eliminate the need for two sets of nearly parallel tests.